src/libAtomVM/mailbox.c

```diff
@@ -342,7 +351,7 @@ Message *mailbox_first(Mailbox *mbox)
     MailboxMessage *msg = mbox->receive_pointer;
     Message *result = NULL;
     if (msg) {
-        result = &msg->body.normal;
+        result = CONTAINER_OF(msg, Message, base);
```

Comment:
Typically this will be simplified by `receive_pointer` being a `Message *`.

Reply:
That change has been reverted.